RepositoryCompiler/CodeModel/CaDETModel/CodeItems/CaDETMember.cs

```diff
@@ -1,43 +1,62 @@
-﻿using System.Collections.Generic;
-using System.Linq;
-using RepositoryCompiler.CodeModel.CaDETModel.Metrics;
+﻿using RepositoryCompiler.CodeModel.CaDETModel.Metrics;
+using System.Collections.Generic;
+using System.Text;
 
 namespace RepositoryCompiler.CodeModel.CaDETModel.CodeItems
 {
     public class CaDETMember
     {
-        public string Name { get; set; }
-        public CaDETMemberType Type { get; set; }
-        public string SourceCode { get; set; }
-        public CaDETClass Parent { get; set; }
-        public List<string> Params { get; set; }
-        public List<CaDETModifier> Modifiers { get; set; }
-        public ISet<CaDETMember> InvokedMethods { get; set; }
-        public ISet<CaDETMember> AccessedFieldsAndAccessors { get; set; }
-        public CaDETMemberMetrics Metrics { get; set; }
+        public string Name { get; internal set; }
+        public CaDETMemberType Type { get; internal set; }
+        public string SourceCode { get; internal set; }
+        public CaDETClass Parent { get; internal set; }
+        public List<CaDETParameter> Params { get; internal set; }
+        public List<CaDETModifier> Modifiers { get; internal set; }
+        public CaDETMemberMetrics Metrics { get; internal set; }
+        public ISet<CaDETMember> InvokedMethods { get; internal set; }
+        public ISet<CaDETMember> AccessedAccessors { get; internal set; }
+        public ISet<CaDETField> AccessedFields { get; internal set; }
 
         public override bool Equals(object other)
         {
             if (!(other is CaDETMember otherMember)) return false;
-            if (Parent == null) return Name.Equals(otherMember.Name);
-            bool nameAndParentEqual = Name.Equals(otherMember.Name) && Parent.Equals(otherMember.Parent);
-            //This is a messy hack. The problem is that this class encapsulates methods/constructors/properties and fields.
-            //Should refactor.
-            if (Type.Equals(CaDETMemberType.Field)) return nameAndParentEqual;
-            return nameAndParentEqual && !Params.Except(otherMember.Params).Any();
+            return Parent.Equals(otherMember.Parent) && GetSignature().Equals(otherMember.GetSignature());
         }
+
+        public string GetSignature()
+        {
+            var sb = new StringBuilder();
+            if (Parent != null) sb.Append(Parent.FullName).Append(".");
+            sb.Append(Name);
+            if (Params != null)
+            {
+                sb.Append("(");
+                for (var i = 0; i < Params.Count; i++)
+                {
+                    sb.Append(Params[i].Type);
+                    if (i < Params.Count - 1) sb.Append(", ");
+                }
+                sb.Append(")");
+            }
+
+            return sb.ToString();
+        }
+
```

Comment:
### Comment

I understand what is sb, but i need few seconds to understand.

### Proposal

Instead of name sb, more readable is maybe signature:

```sh
        public string GetSignature()
        {
            var signature = new StringBuilder();
            if (Parent != null) signature.Append(Parent.FullName).Append(".");
            signature.Append(Name);
            if (Params != null)
            {
                signature.Append("(");
                for (var i = 0; i < Params.Count; i++)
                {
                    signature.Append(Params[i].Type);
                    if (i < Params.Count - 1) signature.Append(", ");
                }
                signature.Append(")");
            }

            return signature.ToString();
        }
```

Reply:
"sb" is often used as a name for an instance of the StringBuilder class. To make it more explicit I'll name it signatureBuilder.